lib/datadog/tracing/contrib/concurrent_ruby/context_composite_executor_service.rb

```diff
@@ -16,16 +16,23 @@ def initialize(composited_executor)
             @composited_executor = composited_executor
           end
 
-          # post method runs the task within composited executor - in a different thread
+          # post method runs the task within composited executor - in a different thread. The original arguments are
+          # captured to be propagated to the composited executor post method
           def post(*args, &task)
             tracer = Tracing.send(:tracer)
             parent_context = tracer.provider.context
+            executor = @composited_executor.is_a?(Symbol) ? Concurrent.executor(@composited_executor) : @composited_executor
 
-            @composited_executor.post(*args) do
+            # Pass the original arguments to the composited executor, which
+            # pushes them (possibly transformed) as block args
+            executor.post(*args) do |*block_args|
               begin
                 original_context = tracer.provider.context
                 tracer.provider.context = parent_context
```

Comment:
I don't think this type of context sharing works as we expect anymore.

It's not safe to share the same context object between different threads. They will fight for who the `active_span` is at any point in time, and span parenting will likely get intertwined between all threads sharing the same context.

We should use TraceDigest propagation instead: https://github.com/DataDog/dd-trace-rb/blob/c6c3935f884f20de76de402dcc2db55da48c978d/lib/datadog/tracing.rb#L23

Reply:
updated to use `continue_trace!`